libs/constrained-generators/src/Constrained/SumList.hs

```diff
@@ -0,0 +1,308 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Constrained.SumList where
+
+import Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NE
+import System.Random (Random (..))
+import Test.QuickCheck (Gen, choose, shuffle, vectorOf)
+
+-- =======================================================
+-- Helper functions for genSizedList
+
+data Solution t = Yes (NonEmpty [t]) | No [String]
+  deriving (Eq)
+
+instance Show t => Show (Solution t) where
+  show (No xs) = "No" ++ "\n" ++ unlines xs
+  show (Yes xs) = "Yes " ++ show xs
+
+-- | Similar to (concat :: [[x]] -> [x]) for lists
+concatNE :: NonEmpty (NonEmpty t) -> NonEmpty t
+concatNE (x :| []) = x
+concatNE (x :| (y : ys)) = x <> concatNE (y :| ys)
+
+partitionNE :: [Solution t] -> ([NonEmpty [t]], [[String]])
+partitionNE xs = go xs ([], [])
+  where
+    go [] (ys, ns) = (reverse ys, reverse ns)
+    go (Yes y : more) (ys, ns) = go more (y : ys, ns)
+    go (No n : more) (ys, ns) = go more (ys, n : ns)
```

Comment:
I suggest reusing existing functionality in `base` for things like that, instead of reimplementing it from scratch. For example current version suffers from a space leak, which is quite easily fixed with [`partitionEither`](https://downloads.haskell.org/ghc/9.10.1/docs/libraries/base-4.20.0.0-1f57/Data-Either.html#v:partitionEithers)
```suggestion
partitionSolutons = 
  partitionEithers . map $ \case
    Yes y -> Left y
    No n -> Right n
```

Also note the change to the name, since `NE` stands for non-empty I assume, which does not seem to be relevant to the function

Reply:
done